### ISSUE TITLE
Please allow class contain memory of other class or  fixed size array

### ISSUE THREAD

**Issue by ygc369** [dotnet/coreclr#735]
Thus, total object number can be less, and GC pressure can be lower, even accessing class fields can be faster because they can be directly located by accessing memory only once, not twice or more.
Look at the following picture which show memory layout.
![image](https://cloud.githubusercontent.com/assets/11640479/7203556/fa5b064e-e54d-11e4-89c2-ccf3869daf18.png)
The class or array included in other class would act as struct, and can't be referenced.
For example, CLR should allow users to define class like this:
class A
{
    class B { int x; char c; };
    struct B b; // for example, use keyword "struct" to force class B to act as struct
    B b0; // a reference to class B
    B b1[10]; // reference array, the array itself is stored in class A, but b1[0]~b1[9] are only reference
    struct B b2[10]; //b2[0]~b2[9] are not reference, they store all fields of class B
    int y[20]; //instead of "unsafe fixed int y[20]", when access "y", the index should be checked, so there is no necessary to use the keyword "unsafe"
};

**Comment by jkotas** [comment]
This is a bag of C# language feature requests. Similar feature requests have actually been topic of recent C# language design discussion - check https://github.com/dotnet/roslyn/labels/Design%20Notes, e.g. https://github.com/dotnet/roslyn/issues/126 is exactly one of your ideas.

I suggest that you look at the ongoing design discussions on C# language design, and open new feature requests for the individual ideas that are not tracked yet. I am going to close this issue because of it is not actionable here.

(BTW: CLR allows you to do most of the above, though using unverifiable IL.)

**Comment by ygc369** [comment]
@jkotas 
Thank you for your reply, I'm sorry for not looking at  the ongoing design discussions on C# language design, but I think this issue shouldn't be closed. Though some of my ideas have been topic, others are not, such as using an existing class as struct. Please add my requests to design discussions on C# language design except those already tracked.
The picture above has someting wrong, please ignore it and look at the following picture.
class A
{
class B { int x; char c; };
struct B b; // for example, use keyword "struct" to force class B to act as struct
B b0; // a reference to class B
B b1[10]; // reference array, the array itself is stored in class A, but b1[0]~b1[9] are only reference
struct B b2[10]; //b2[0]~b2[9] are not reference, they store all fields of class B
B[] b3; //a reference to a reference array
struct B[] b4; //a reference to an array which stores values directly
int y[20]; //instead of "unsafe fixed int y[20]", when access "y", the index should be checked, so there is no necessary to use the keyword "unsafe"
int[] z; //normal array
};
![mem](https://cloud.githubusercontent.com/assets/11640479/7213687/7eda7b32-e5b9-11e4-8c7e-66b2a9496136.png)
If these requests can be realized, C# can be as flexible as C/C++.

**Comment by kangaroo** [comment]
@ygc369 Hi! So @jkotas isn't suggesting your issue shouldn't be addressed.  He's (rightfully) pointing out its filed in the wrong place.  This repository is for the _runtime_ which already supports the type of features you want.  You are asking for language extensions, so they should likely be filed in the roslyn repository and tracked by that team.

This doesn't say the issue or ideas aren't good, simply that they're in the wrong place :)
